File: Code_Test/cpp/dfs_bfs/5-2queue.cpp
#include <iostream>
#include <queue>

std::queue<int> q; // different from deque of python3

int main(){
    // 삽입(5) - 삽입(2) - 삽입(3) - 삽입(7) - 삭제() - 삽입(1) - 삽입(4) - 삭제()
    q.push(5);
    q.push(2);
    q.push(3);
    q.push(7);
    q.pop(); // delete the first element
    q.push(1);
    q.push(4);
    q.pop();

    // 먼저 들어온 원소부터 추출
    while (!q.empty()) {
        std::cout << q.front() << ' ';
        q.pop();
    }
}

File: Code_Test/py/dfs_bfs/5-2queue.py
from collections import deque
from turtle import left # 가장 앞에 삽입 삭제 하는데 O(1) vs list는 O(N)
# FIFO (like the lines of CAVA)
queue = deque()

queue.append(5)
queue.append(2)
queue.append(3)
queue.append(7)
queue.popleft() # delete the first element # appendleft 하면 가장 앞에 원소 삽입 가능

queue.append(1)
queue.append(4)
queue.popleft() # delete the first element
queue.pop() # delete the last(recent) element
queue.appendleft(9) # insert 9 at the first

print(queue)
queue.reverse()
print(queue)
